core-backend/sandbox/src/funcs.rs

```diff
@@ -595,15 +596,16 @@ where
 
         let dest = pop_i32(&mut args)?;
 
-        let maybe_message_id = ctx
+        let opt_details = ctx
             .ext
-            .with_fallible(|ext| ext.reply_to().map_err(FuncError::Core))
+            .with_fallible(|ext| ext.reply_details().map_err(FuncError::Core))
             .map_err(|err| {
                 ctx.err = err;
                 HostError
             })?;
 
-        if let Some((message_id, _)) = maybe_message_id {
+        if let Some(details) = opt_details {
+            let message_id = details.into_parts().0;
```

Comment:
Also, do we really need `into_parts` as parts are copy types?
```suggestion
            let message_id = details.reply_to();
```

Reply:
I don't see the reason to clone/copy even copy-types when there is no need to access details further. Move-semantic makes sense for me even in such small things.
Isn't this destruction cheaper then copy?

Reply:
We can provide something like
`fn into_reply_to(self) -> MessageId` for comfortable usage